streams/src/test/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreatorTest.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.kafka.streams.state.internals.ThreadCache;
+import org.apache.kafka.test.MockClientSupplier;
+import org.easymock.EasyMockRunner;
+import org.easymock.Mock;
+import org.easymock.MockType;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.easymock.EasyMock.anyString;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.mock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(EasyMockRunner.class)
+public class ActiveTaskCreatorTest {
+
+    @Mock(type = MockType.NICE)
+    private InternalTopologyBuilder builder;
+    @Mock(type = MockType.NICE)
+    private StreamsConfig config;
+    @Mock(type = MockType.NICE)
+    private StateDirectory stateDirectory;
+    @Mock(type = MockType.NICE)
+    private ChangelogReader changeLogReader;
+    @Mock(type = MockType.NICE)
+    private Consumer<byte[], byte[]> consumer;
+    @Mock(type = MockType.NICE)
+    private Admin adminClient;
+
+    private final MockClientSupplier mockClientSupplier = new MockClientSupplier();
+    final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(new Metrics(), "clientId", StreamsConfig.METRICS_LATEST);
+
+    private ActiveTaskCreator activeTaskCreator;
+
+    @Test
+    public void shouldConstructProducerMetricsWithoutEOS() {
+        expect(config.getString(StreamsConfig.APPLICATION_ID_CONFIG)).andReturn("appId");
+        expect(config.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG)).andReturn(StreamsConfig.AT_LEAST_ONCE);
+        expect(config.getProducerConfigs(anyString())).andReturn(Collections.emptyMap());
+        replay(config);
+
+        activeTaskCreator = new ActiveTaskCreator(
+            builder,
+            config,
+            streamsMetrics,
+            stateDirectory,
+            changeLogReader,
+            new ThreadCache(new LogContext(), 0L, streamsMetrics),
+            new MockTime(),
+            mockClientSupplier,
+            "threadId",
+            new LogContext().logger(ActiveTaskCreator.class)
+        );
+
+        final MetricName testMetricName = new MetricName("test_metric", "", "", new HashMap<>());
+        final Metric testMetric = new KafkaMetric(
+            new Object(),
+            testMetricName,
+            (Measurable) (config, now) -> 0,
+            null,
+            new MockTime());
+
+        mockClientSupplier.producers.get(0).setMockMetrics(testMetricName, testMetric);
+        final Map<MetricName, Metric> producerMetrics = activeTaskCreator.producerMetrics();
+        assertEquals(testMetricName, producerMetrics.get(testMetricName).metricName());
+    }
+
+    @Test
+    public void shouldConstructProducerMetricsWithEOS() {
```

Comment:
Could we use `verify` on mocks for both test cases?

Reply:
We don't really care what methods get called, do we? We are only interesting in the result of the method call -- if the internal implementation on how we build the metric change, the test should still pass.